streams_map_incoming_bidi.go

```diff
@@ -159,12 +159,15 @@ func (m *incomingBidiStreamsMap) deleteStream(num protocol.StreamNum) error {
 	delete(m.streams, num)
 	// queue a MAX_STREAM_ID frame, giving the peer the option to open a new stream
 	if m.maxNumStreams > uint64(len(m.streams)) {
-		numNewStreams := m.maxNumStreams - uint64(len(m.streams))
-		m.maxStream = m.nextStreamToOpen + protocol.StreamNum(numNewStreams) - 1
-		m.queueMaxStreamID(&wire.MaxStreamsFrame{
-			Type:         protocol.StreamTypeBidi,
-			MaxStreamNum: m.maxStream,
-		})
+		maxStream := m.nextStreamToOpen + protocol.StreamNum(m.maxNumStreams-uint64(len(m.streams))) - 1
+		// Never send a value larger than protocol.MaxStreamCount.
+		if maxStream <= protocol.MaxStreamCount {
```

Comment:
In the false case, should we send a capped maxStreamId frame, error, …? Silently ignoring this seems fragile?

Reply:
There's no error here. It's just that we're running out of stream IDs (which are limit by the way they're encoded... to 2^60 per stream type).